src/Soil-Core/SoilIndexIterator.class.st

```diff
@@ -579,6 +579,15 @@ SoilIndexIterator >> removeKey: key value: value ifAbsent: aBlock [
 	^ oldValue
 ]
 
+{ #category : #removing }
+SoilIndexIterator >> removeKeyAll: key ifAbsent: aBlock [
+	"we remove all the entries for key"
```

Comment:
as aBlock is not used it pretty much shows it is not needed ;) If we remove all keys we can just ignore if none is there. The declaration of having no key after is met